src/connected/autobinding.ts

```diff
@@ -156,4 +165,161 @@ export class AutoBindingService {
         return false;
       });
   }
+
+  private async getBestHitsForConnections(connectionToServerProjects: Map<BaseConnection, ServerProject[]>,
+                                          unboundFolder: VSCode.WorkspaceFolder):
+    Promise<Map<BaseConnection, MatchHit[]>> {
+    const folderName = unboundFolder.name;
+    const workspaceName = VSCode.workspace.name;
+    const folderNameTokens = [...tokenizeString(folderName)];
+    const workspaceNameTokens = [...tokenizeString(workspaceName)];
+    const connectionToBestHits = new Map<BaseConnection, MatchHit[]>();
+    for (const [connection, projects] of connectionToServerProjects) {
+      let bestHits: MatchHit[] = [];
+      bestHits.push({ hits: 0, projectKey: '', connection: { connectionId: '' } });
+      for (const project of projects) {
+        const projectKey = project.key;
+        const projectName = project.name;
+        const serverProjectString = (projectKey + projectName).toLowerCase();
+        const folderNameHits = this.getHits(folderNameTokens, serverProjectString);
+        const workspaceNameHits = this.getHits(workspaceNameTokens, serverProjectString);
+        const bestHitCount = bestHits[0].hits;
+        if (folderNameHits >= bestHitCount || workspaceNameHits >= bestHitCount) {
+          bestHits = this.updateBestHits(bestHits, folderNameHits, workspaceNameHits,
+            projectKey, connection);
+        }
+      }
+      if (bestHits[0].hits > 0) {
+        connectionToBestHits.set(connection, bestHits);
+      }
+    }
+    return connectionToBestHits;
+  }
+
+  private async promptToAutoBind(connectionToBestHits: Map<BaseConnection, MatchHit[]>,
+                                 unboundFolder: VSCode.WorkspaceFolder) {
+    const [bestHits] = connectionToBestHits.values();
+    if (connectionToBestHits.size === 1 && bestHits.length === 1) {
+      const bestHit = bestHits[0];
+      this.promptToAutoBindSingleOption(bestHit, connectionToBestHits, unboundFolder);
+    } else {
+      this.promptToAutoBindMultiChoice(connectionToBestHits, unboundFolder);
+    }
+  }
+
+  private async promptToAutoBindSingleOption(bestHit: MatchHit, connectionToBestHits: Map<BaseConnection,
+    MatchHit[]>, unboundFolder: VSCode.WorkspaceFolder) {
+    const [connection] = connectionToBestHits.keys();
+    const connectionName = getDisplayName(connection);
+    const result = await VSCode.window.showInformationMessage(
+      `There is a project to bind on ${connectionName}. Do you want to bind?`, 'Bind');
```

Comment:
I think it would be useful to add the matching server project name in the message. Something like
`There is a project ${bestHit.projectKey} on ${connectionName}. Do you want to bind?`